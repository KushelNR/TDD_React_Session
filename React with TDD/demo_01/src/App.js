import React, { Component } from 'react';
import { Button } from 'react-bootstrap';

export default class App extends Component {
    constructor() {
        super();

        this.state = {
            todos: []
        }
    }

    onClick = () => {
        const { todos } = this.state;

        let ids = todos.map(t => t.id);

        let max_id = ids.length > 0 ? Math.max(...ids) : 0;

        todos.push({ id: max_id + 1 });

        this.setState({ todos });
    }


    render() {
        const { todos } = this.state;

        return (
            <div>
                <h1>TODOS</h1>

                <div className='todos'>
                    {todos.map(todo => {
                        return (
                            <div key={todo.id}>{todo.id}</div>
                        );
                    })}
                </div>
                
                <Button className='add-btn' onClick={this.onClick}>Add Todo</Button>
            </div>
        )
    }
}
