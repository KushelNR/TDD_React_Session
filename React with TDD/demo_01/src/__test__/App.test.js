import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

let app = shallow(<App debug />);

describe('App', () => {
    it('Should render correctly in `debug`', () => {
        expect(app).toMatchSnapshot();
    });

    it('Init the `state` with default values', () => {
        expect(app.state().todos).toEqual([]);
    });

    describe('When clicking the `add todo` button,', () => {
        beforeEach(() => {
            app.find('.add-btn').simulate('click');
        });

        afterEach(() => {
            app.setState({ todos: [] });
        });

        it('A todo item should add to todos array in the state', () => {
            expect(app.state().todos).toEqual([{ id: 1 }]);
        });
    
        it('Add todo item to todos list in view', () => {
            expect(app.find('.todos').children().length).toEqual(1);
        });

        it('Add another todo item to todos state', () => {
            app.find('.add-btn').simulate('click');

            expect(app.state().todos.length).toEqual(2);
        });
    });
});