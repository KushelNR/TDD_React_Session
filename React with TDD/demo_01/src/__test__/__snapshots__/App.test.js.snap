// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render correctly in \`debug\` 1`] = `
<div>
  <h1>
    TODOS
  </h1>
  <div
    className="todos"
  />
  <Button
    active={false}
    className="add-btn"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Add Todo
  </Button>
</div>
`;
